test/testactivitylistmodel.cpp

```diff
@@ -0,0 +1,408 @@
+                data.append(_activityData[dataIndex]);
+            }
+        }
+
+        QJsonObject root;
+        QJsonObject ocs;
+        ocs.insert(QStringLiteral("data"), data);
+        root.insert(QStringLiteral("ocs"), ocs);
+
+        return QJsonDocument(root).toJson();
+    }
+
+private:
+    static FakeRemoteActivityStorage *_instance;
+    QJsonArray _activityData;
+    QVariantMap _metaSuccess;
+    quint32 _numItemsToInsert = 30;
+    int _startingId = STARTING_ID;
+};
+
+FakeRemoteActivityStorage *FakeRemoteActivityStorage::_instance = nullptr;
+
+class TestActivityListModel : public QObject
+{
+    Q_OBJECT
+
+public:
+    TestActivityListModel() = default;
+    ~TestActivityListModel() override {
+        OCC::AccountManager::instance()->deleteAccount(accountState.data());
+    }
+
+    QScopedPointer<FakeQNAM> fakeQnam;
+    OCC::AccountPtr account;
+    QScopedPointer<OCC::AccountState> accountState;
+    QScopedPointer<TestingALM> model;
+    QScopedPointer<QAbstractItemModelTester> modelTester;
+
+    OCC::Activity testNotificationActivity;
+
+    static const int searchResultsReplyDelay = 100;
+
+private slots:
+    void initTestCase()
+    {
+        fakeQnam.reset(new FakeQNAM({}));
+        account = OCC::Account::create();
+        account->setCredentials(new FakeCredentials{fakeQnam.data()});
+        account->setUrl(QUrl(("http://example.de")));
+
+        accountState.reset(new OCC::AccountState(account));
+
+        fakeQnam->setOverride([this](QNetworkAccessManager::Operation op, const QNetworkRequest &req, QIODevice *device) {
+            Q_UNUSED(device);
+            QNetworkReply *reply = nullptr;
+
+            const auto urlQuery = QUrlQuery(req.url());
+            const auto format = urlQuery.queryItemValue(QStringLiteral("format"));
+            const auto since = urlQuery.queryItemValue(QStringLiteral("since")).toInt();
+            const auto limit = urlQuery.queryItemValue(QStringLiteral("limit")).toInt();
+            const auto path = req.url().path();
+
+            if (!req.url().toString().startsWith(accountState->account()->url().toString())) {
+                reply = new FakeErrorReply(op, req, this, 404, fake404Response);
+            }
+            if (format != QStringLiteral("json")) {
+                reply = new FakeErrorReply(op, req, this, 400, fake400Response);
+            }
+
+            if (path.startsWith(QStringLiteral("/ocs/v2.php/apps/activity/api/v2/activity"))) {
+                reply = new FakePayloadReply(op, req, FakeRemoteActivityStorage::instance()->activityJsonData(since, limit), searchResultsReplyDelay, fakeQnam.data());
+            }
+
+            if (!reply) {
+                return qobject_cast<QNetworkReply*>(new FakeErrorReply(op, req, this, 404, QByteArrayLiteral("{error: \"Not found!\"}")));
+            }
+
+            return reply;
+        });
+
+        model.reset(new TestingALM(accountState.data()));
```

Comment:
why pointers and not directly declaring an instance of `TestingALM` in class `TestActivityListModel` ?
I would prefer not to use pointers when we can avoid even in automated tests code

Reply:
Done; the `model` declaration is now just a `TestingALM`, not a pointer to it